src/sock.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2012-2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <compat.h>
+#include <netaddress.h>
+#include <netbase.h>
```

Comment:
```suggestion
#include <netbase.h> // For CloseSocket
```

It seems this is the only Bitcoin Core related dependency. Are the other includes needed at all? If not, it would be good to move this module to `util`, which is the place for general utility stuff (not bitcoin related).

Reply:
No, the other includes are not needed (and I removed them). However there is a circular dependency now:

* netbase -> sock (`netbase.cpp` uses the `Sock` class, of course)
* sock -> netbase (`sock.cpp` uses `MillisToTimeval()` and `CloseSocket()` which are defined in `netbase.cpp`)

One way to resolve that is to put the `Sock` class in `netbase.{h,cpp}` and ditch `sock.{h,cpp}`.

Another one would be to move `MillisToTimeval()` and `CloseSocket()` outside of `netbase.cpp`, somewhere in `src/util/` and use them from both `netbase.cpp` and from `sock.cpp`.

Hmm...

Reply:
> One way to resolve that is to put the `Sock` class in `netbase.{h,cpp}`...

Did that.

Reply:
This was finally [changed](https://github.com/bitcoin/bitcoin/pull/20788#issuecomment-756754017) to:

> move MillisToTimeval() and CloseSocket() outside of netbase.cpp, somewhere in src/util/ and use them from both netbase.cpp and from sock.cpp

it looks much better now.